### ISSUE TITLE
Plots.jl shorthands

### ISSUE THREAD

**Issue by asinghvi17** [JuliaPlots/Makie.jl#368]
Plots.jl has a lot of small, easy-to-implement convenience functions and recipes that would be nice to have in AbstractPlotting, or StatsMakie.

- [ ] https://github.com/JuliaPlots/Plots.jl/blob/e061d156bc0a826db1d8a53d89cd8a2852f806e2/src/shorthands.jl#L70-L77

- [ ] https://github.com/JuliaPlots/Plots.jl/blob/e061d156bc0a826db1d8a53d89cd8a2852f806e2/src/shorthands.jl#L79-L86

- [ ] https://github.com/JuliaPlots/Plots.jl/blob/e061d156bc0a826db1d8a53d89cd8a2852f806e2/src/shorthands.jl#L139-L151

- [ ] https://github.com/JuliaPlots/Plots.jl/blob/e061d156bc0a826db1d8a53d89cd8a2852f806e2/src/shorthands.jl#L139-L151

- [ ] https://github.com/JuliaPlots/Plots.jl/blob/e061d156bc0a826db1d8a53d89cd8a2852f806e2/src/shorthands.jl#L166-L178

- [ ] https://github.com/JuliaPlots/Plots.jl/blob/e061d156bc0a826db1d8a53d89cd8a2852f806e2/src/shorthands.jl#L180-L192

- [ ] https://github.com/JuliaPlots/Plots.jl/blob/e061d156bc0a826db1d8a53d89cd8a2852f806e2/src/shorthands.jl#L224-L239

- [ ] https://github.com/JuliaPlots/Plots.jl/blob/e061d156bc0a826db1d8a53d89cd8a2852f806e2/src/shorthands.jl#L368-L380

- [ ] https://github.com/JuliaPlots/Plots.jl/blob/e061d156bc0a826db1d8a53d89cd8a2852f806e2/src/shorthands.jl#L382-L384

**Comment by asinghvi17** [comment]
Implementation for sticks (needs to be turned into a recipe but you get the picture):

```julia
function sticks!(scene, x, y)
for (i, j) in zip(x, y)
    linesegments!(scene, (Point2f0(i, min(j, 0)), Point2f0(i, max(j, 0)))
end
end
```
![sticks-makie](https://user-images.githubusercontent.com/32143268/60091389-e7e7ed80-975d-11e9-9a65-7a0d9f7d3db7.png)

**Comment by asinghvi17** [comment]
Here's GR.jl's axis drawing code, for reference.

https://github.com/jheinen/GR.jl/blob/9edecc09594ac855bae70113a6fd452db12c5ea0/src/jlgr.jl#L414-L526
```julia
function draw_polar_axes()
    viewport = plt.kvs[:viewport]
    diag = sqrt((viewport[2] - viewport[1])^2 + (viewport[4] - viewport[3])^2)
    charheight = max(0.018 * diag, 0.012)


    window = plt.kvs[:window]
    rmin, rmax = window[3], window[4]


    GR.savestate()
    GR.setcharheight(charheight)
    GR.setlinetype(GR.LINETYPE_SOLID)


    tick = 0.5 * GR.tick(rmin, rmax)
    n = round(Int, (rmax - rmin) / tick + 0.5)
    for i in 0:n
        r = float(i) / n
        if i % 2 == 0
            GR.setlinecolorind(88)
            if i > 0
                GR.drawarc(-r, r, -r, r, 0, 359)
            end
            GR.settextalign(GR.TEXT_HALIGN_LEFT, GR.TEXT_VALIGN_HALF)
            x, y = GR.wctondc(0.05, r)
            GR.text(x, y, string(signif(rmin + i * tick, 12)))
        else
            GR.setlinecolorind(90)
            GR.drawarc(-r, r, -r, r, 0, 359)
        end
    end
    for alpha in 0:45:315
        a = alpha + 90
        sinf = sin(a * π / 180)
        cosf = cos(a * π / 180)
        GR.polyline([sinf, 0], [cosf, 0])
        GR.settextalign(GR.TEXT_HALIGN_CENTER, GR.TEXT_VALIGN_HALF)
        x, y = GR.wctondc(1.1 * sinf, 1.1 * cosf)
        GR.textext(x, y, string(alpha, "^o"))
    end
    GR.restorestate()
end

**Comment by asinghvi17** [comment]
Plots.jl bezier curves:
https://github.com/JuliaPlots/Plots.jl/blob/2816b1128f2d2b2fdba5b170facd316954eb61e2/src/recipes.jl#L280-L339

**Comment by asinghvi17** [comment]
For `vline` and `hline`, is there a way to get the Scene's limits as an Observable (or, in general, a way to manipulate the Scene within the plotting command that doesn't involve bypassing the recipe pipeline and creating a function `vline!(scene, xcoord; linesegment_kwargs...)`)?

**Comment by asinghvi17** [comment]
@jheinen, this doesn't seem to be in the GR documentation - what does `gr_ wctondc` (from the C-API) do?

**Comment by jheinen** [comment]
`gr_wctondc` converts the user's "world coordinates" to "normalized device coordinates" [0, 1]

**Comment by asinghvi17** [comment]
Thanks!  Will look into the polar axis adaptation further.

**Comment by ym-han** [comment]
Hi, could I ask what the progress on this has been like / what the status of this is? I would be interested in helping out if you need help --- I'm guessing that it won't be very hard or require metaprogramming skills (though please correct me if I'm wrong)!

**Comment by TabeaW** [comment]
I have done a bit of work for a hexbin plot (everything I had imagined should work fine except the labelling for the colorbar with the logscale -> image: bottom right) . I am not totally sure if I had done everything right with the Observables, is there an easy way to test this?
![image](https://user-images.githubusercontent.com/35938558/144262329-f1a62849-c18c-4160-95a4-833a7575f962.png)

**Comment by SimonDanisch** [comment]
That looks great :) 

>is there an easy way to test this?

Make an animation with an observable? Or what do you mean?

**Comment by TabeaW** [comment]
Yes, exactly. I tried it, but maybe I mixed something up with the observables because I get an error  `LoadError: Metadata array needs to have same length as data.
                    Found 7218 data items, and 4650 metadata items`

**Comment by TabeaW** [comment]
Nevermind, managed it, should work now :)

**Comment by jkrumbiegel** [comment]
Very cool, is this going to be a PR?

**Comment by Azzaare** [comment]
Hi there!

I wonder if there is any progress regarding `areaplot`

I can try to help if someone gives me a process on how to start (should I try to port things from plots.jl?)

**Comment by SimonDanisch** [comment]
I guess one can do that with: `band(1:10, 0, rand(10))`?

**Comment by asinghvi17** [comment]
Yeah pretty much, since one can get the transformed plot limits from the model matrix theoretically it could be done in a recipe.  However, for the best implementation this should probably wait until we have a better interface for using Axis-level attributes in recipes, so that the palette can be shared.

**Comment by Azzaare** [comment]
I see. I will try to use a workaround with `band` while waiting for the official recipe!

Thanks

**Comment by Azzaare** [comment]
Just for the follow-up, the second `Axis` in the video is the result of using `band`. Thanks again. 

I assume you don't need a MWE, but I can write one to illustrate the areaplot recipe for Makie.

https://user-images.githubusercontent.com/4535131/181269095-914f417a-4349-4de5-8eda-ce22b5451e5f.mp4

**Comment by SimonDanisch** [comment]
A PR with the recipe would be even greater ;)

**Comment by Moelf** [comment]
hexbin has been added?
